giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/delete/impl/DeleteDocumentService.java

```diff
@@ -0,0 +1,151 @@
+import edu.asu.diging.gilesecosystem.web.core.service.core.ITransactionalProcessingRequestService;
+import edu.asu.diging.gilesecosystem.web.core.service.core.ITransactionalUploadService;
+import edu.asu.diging.gilesecosystem.web.core.service.delete.IDeleteDocumentService;
+import edu.asu.diging.gilesecosystem.web.core.service.properties.Properties;
+
+/**
+ * This class handles the deletion of a document.
+*/
+
+@Service
+public class DeleteDocumentService implements IDeleteDocumentService {
+    
+    public final static String REQUEST_PREFIX = "DELREQ";
+    
+    @Autowired
+    private IRequestFactory<IStorageDeletionRequest, StorageDeletionRequest> requestFactory;
+    
+    @Autowired
+    private ApplicationContext ctx;
+    
+    @Autowired
+    private ISystemMessageHandler messageHandler;
+    
+    @Autowired
+    private IRequestProducer requestProducer;
+    
+    @Autowired
+    private IPropertiesManager propertyManager;
+    
+    @Autowired
+    private ITransactionalFileService fileService;
+    
+    @Autowired
+    private ITransactionalUploadService uploadService;
+    
+    @Autowired
+    private ITransactionalDocumentService documentService;
+    
+    @Autowired
+    private ITransactionalProcessingRequestService processingRequestService;
+    
+    private Map<String, FileType> fileTypes;
+    
+    @PostConstruct
+    public void init() {
+        fileTypes = new HashMap<String, FileType>();
+        
+        // register what content types should be treated as what file type
+        Map<String, IFileTypeHandler> ctxMap = ctx.getBeansOfType(IFileTypeHandler.class);
+        Iterator<Entry<String, IFileTypeHandler>> iter = ctxMap.entrySet().iterator();
+        
+        while(iter.hasNext()){
+            Entry<String, IFileTypeHandler> handlerEntry = iter.next();
+            IFileTypeHandler handler = (IFileTypeHandler) handlerEntry.getValue();
+            for (String type : handler.getHandledFileTypes()) {
+                fileTypes.put(type, handler.getHandledFileType());
+            }
+        }
+        requestFactory.config(StorageDeletionRequest.class);
+    }
+    
+    @Override
+    @Async
+    public void initiateDeletion(IDocument document) throws GilesProcessingException, MessageCreationException {
+        IRequest storageDeletionRequest = createRequest(document);
+        requestProducer.sendRequest(storageDeletionRequest, getTopic());
+    }
+    
+    @Override
+    public void completeDeletion(IDocument document) {
+        deleteFiles(document);
+        deleteProcessingRequests(document);
+        documentService.deleteDocument(document.getId());
+        deleteUpload(document.getUploadId());
+    }
+    
+    private void deleteProcessingRequests(IDocument document) {
+        processingRequestService.deleteRequestsByDocumentId(document.getId());
+    }
+    
+    private void deleteFiles(IDocument document) {
+        fileService.deleteFiles(document.getId());
+    }
```

Comment:
are these two methods above really needed? It's just one-liners and the methods are private.